File: Basic Programming/Hackerrank 30 days of code c++ solution/Day_9.cpp
/*                    Name : Arif Khan
                      Judge: HACKERRANK
                      University: Primeasia University
                      problem: Day 9: Recursion 3
                      Difficulty: Easy
                      Problem Link: https://www.hackerrank.com/challenges/30-recursion/problem
*/


#include <bits/stdc++.h>
using namespace std;

int factorial(int n);
int main() {
    int n1,r;
    cin >> n1;
    r = factorial(n1);
    cout << r <<endl;
    return 0;
}
int factorial(int n){
    if(n<=1)
        return 1;
    else 
        return n*factorial(n-1);
}

File: Basic Programming/Hackerrank 30 days of code python3 solution/Day_9.py
#                      Name : Arif Khan
#                      Judge: HACKERRANK
#                      University: Primeasia University
#                      problem: Day 9: Recursion 3
#                      Difficulty: Easy
#                      Problem Link: https://www.hackerrank.com/challenges/30-recursion/problem
#

import sys

def factorial(n):
    if(n<=1):
        return 1
    else:
        return factorial(n-1)*n

if __name__ == "__main__":
    n = int(input().strip())
    result = factorial(n)
    print(result)